File: backend/models/Experience.js
const mongoose = require('mongoose');

const experienceSchema = new mongoose.Schema({
    user_id: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true, index: true },
    company: { type: String, required: true },
    role: { type: String, required: true },
    description: { type: String, required: true },
    technologies: { type: String, default: null },
    location: { type: String, default: null },
    is_current: { type: Boolean, default: false },
    start_date: { type: Date, default: null },
    end_date: { type: Date, default: null },
}, { timestamps: { createdAt: 'created_at', updatedAt: 'updated_at' } });

module.exports = mongoose.model('Experience', experienceSchema);

File: python-backend/app/models/experience.py
import uuid
from datetime import datetime, date
from sqlalchemy import Column, String, DateTime, ForeignKey, Text, Date, Boolean
from sqlalchemy.orm import relationship
from app.database import Base


class Experience(Base):
    __tablename__ = "experiences"

    id = Column(String, primary_key=True, default=lambda: str(uuid.uuid4()))
    user_id = Column(String, ForeignKey("users.id", ondelete="CASCADE"), nullable=False, index=True)
    company = Column(String(255), nullable=False)
    role = Column(String(255), nullable=False)
    description = Column(Text, nullable=False)
    technologies = Column(Text, nullable=True)  # Comma-separated
    location = Column(String(255), nullable=True)
    is_current = Column(Boolean, default=False)
    start_date = Column(Date, nullable=True)
    end_date = Column(Date, nullable=True)
    created_at = Column(DateTime, default=datetime.utcnow)

    user = relationship("User", back_populates="experiences")
